libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -152,13 +153,13 @@ void IRGenerationContext::addToInternalDispatch(FunctionDefinition const& _funct
 	solAssert(functionType, "");
 
 	YulArity arity = YulArity::fromType(*functionType);
-
-	if (m_internalDispatchMap.count(arity) != 0 && m_internalDispatchMap[arity].count(&_function) != 0)
-		// Note that m_internalDispatchMap[arity] is a set with a custom comparator, which looks at function IDs not definitions
-		solAssert(*m_internalDispatchMap[arity].find(&_function) == &_function, "Different definitions with the same function ID");
-
-	m_internalDispatchMap[arity].insert(&_function);
-	enqueueFunctionForCodeGeneration(_function);
+	auto it = ranges::find(m_internalDispatchMap[arity], &_function);
+	if (it != ranges::end(m_internalDispatchMap[arity]))
+		solAssert(*it == &_function, "Different definitions with the same function ID");
```

Comment:
This assert is ineffective now. It will never fail:
- if we have the same exact `FunctionDefinition` object, we'll enter `else`.
- if we have a different `FunctionDefinition` that for some wild reason happens to share the ID, the assert will succeed because `find()` will never find that definition (it's no longer a set with a comparator so the comparison is between pointers).

I'd remove it from here and instead add a sanity check in `IRGenerator::generateInternalDispatchFunctions()`. That function used to be able to assume no duplicates because it was receiving a set. Now that it gets a queue, it should assert that items in it have unique IDs.

Reply:
Ops, indeed. I added an assertion there to check if a function definition with same ID is processed more then once.